src/NHibernate/Tuple/DynamicMapInstantiator.cs

```diff
@@ -48,7 +50,14 @@ public object Instantiate()
 
 		protected virtual IDictionary GenerateMap()
 		{
-			return new Hashtable();
+			if ((this.component != null) && (typeof(IDictionary<string, object>).IsAssignableFrom(this.component.Type.ReturnedClass) == true))
+			{
+				return new Dictionary<string, object>();
+			}
+			else
+			{
+				return new Hashtable();
+			}
```

Comment:
If we want to default to strong typed dictionary, we should invert the null logic here:

	if (this.component == null ||
		typeof(IDictionary<string, object>).IsAssignableFrom(this.component.Type.ReturnedClass)
	{
		return new Dictionary<string, object>();
	}
	else
	{
		return new Hashtable();
	}